docs/dart/generating-code.md

```diff
@@ -0,0 +1,174 @@
+logic. For the client, there really isn't anything more to do — the engineer
+can simply call the client methods, rely on the generated types for
+compile-time type-safety and serialization, and focus on the application logic.
+
+## Remote plugins
+
+> Note: The example in the [tutorial](getting-started.md) covers much of this
+> section's content.
+
+[Protobuf plugins][available-plugins] are executables that accept `.proto`
+file inputs and generate various outputs (`.dart` files in this case).
+Performing generation on a remote machine makes local setup easier
+and allows the generation to take place in an isolated
+environment. We'll use [Buf][buf], a modern replacement for
+Google's protobuf compiler, along with [remote plugins][remote-plugins].
+
+This requires installing [Buf's CLI][buf-cli]:
+
+```bash
+brew install bufbuild/buf/buf
+```
+
+When developing a new project, 2 new files need to be created:
+
+- [`buf.yaml`][buf.yaml]
+- [`buf.gen.yaml`][buf.gen.yaml]
+
+The first file, `buf.yaml`, can be created by running:
+
+```bash
+buf config init
+```
+
+The second file, `buf.gen.yaml`, needs to be created manually and specifies
+which plugins should be used to generate code. An example of this
+file is shown below:
+
+```yaml
+version: v2
+plugins:
+  - remote: buf.build/connectrpc/dart
+    out: lib/gen
+  - remote: buf.build/protocolbuffers/dart
+    out: lib/gen
+    include_wkt: true
+    include_imports: true
+```
+
+This file specifies that the [`connect-dart` plugin][connect-dart-plugin]
+should be invoked with the options in `opt`, and that its outputs should be
+placed in the `lib/gen` directory. This plugin is responsible for generating
+`.connect.*.dart` files which contain Dart types derived from the
+defined `service` and `rpc` types in Protobuf files.
+
+The config also includes the [`protocolbuffers/dart` plugin][dart-protobuf-plugin] with
+another set of options that place its `.pb*.dart` outputs
+in the same `lib/gen` directory. This plugin
+generates classes from Protobuf types such as `message` and `enum`.
+
+Together, the two plugins generate all the code that you'll need.
+
+> Details on configuring plugins in `buf.gen.yaml` may be found in
+> [the documentation][remote-plugins], and you can browse the full list of
+> available remote plugins [here][available-plugins].
+
+With these configuration files in place, you can now generate code:
+
+```bash
+buf generate
+```
+
+Given the above config and example `eliza.proto` file, you should now see some
+generated Swift files in the `Generated` directory:
+
+```
+lib/gen
+    ├── eliza.connect.spec.dart
+    ├── eliza.connect.client.dart
+    ├── eliza.pb.swift
+    ├── eliza.pbenum.swift
+    ├── eliza.pbjson.swift
+    └── eliza.pbserver.dart
+```
+
+## Local generation
+
+The [`connect-dart`][connect-dart-plugin] plugin are regular
+Protobuf plugins which can be used with `protoc` and `buf` to generate
```

Comment:
You have a mix of singular and plural here. Not sure if you meant to mention the `protocolbuffers/dart` plugin also?